internal/gen/gen.go

```diff
@@ -109,9 +129,8 @@ func (g *Generator) generateTools(data TemplateData) error {
 	return g.writeFile(buf, "tools")
 }
 
-func (g *Generator) generateServer() error {
-	fileExists := fileExists(fmt.Sprintf("%s/main.go", g.options.OutputDir))
-	if fileExists {
+func (g *Generator) generateServer(data TemplateData) error {
+	if g.options.OnlyTools {
 		return nil
 	}
```

Comment:
Could you re-add the fileExists check after the check for only tools? As this will prevent the server to be re-generated (this is where you would setup authentication and other things to make the MCP server actually work with your graphql api).